app/screens/InfoScreen/SponsorCard.tsx

```diff
@@ -6,29 +6,27 @@ import { translate } from "../../i18n"
 import { openLinkInBrowser } from "../../utils/openLinkInBrowser"
 import { SCREEN_CONTENT_WIDTH } from "../../components/carousel/constants"
 
-export type CommonProps = {
+type CommonProps<T> = {
   containerStyle?: ViewStyle
+  tier: T
 }
 
-export type TierLevelProps =
-  | {
-      externalURL: string
-      logo: { uri: string }
-      promoSummary: string
-      sponsor: string
-      sponsorImages?: never
-      tier: "Platinum" | "Gold"
-    }
-  | {
-      externalURL: never
-      logo?: never
-      promoSummary?: never
-      sponsor?: never
-      sponsorImages: { uri: string; sponsor: string; externalURL?: string }[]
-      tier: "Silver" | "Bronze"
-    }
+type TierOneLevelProps<T> = CommonProps<T> & {
+  externalURL: string
+  logo: { uri: string }
+  promoSummary: string
+  sponsor: string
+}
+
+type TierTwoLevelProps<T> = CommonProps<T> & {
+  sponsorImages: { uri: string; sponsor: string; externalURL?: string }[]
+}
 
-export type SponsorCardProps = CommonProps & TierLevelProps
+type SponsorCardProps =
+  | TierOneLevelProps<"Platinum">
+  | TierOneLevelProps<"Gold">
+  | TierTwoLevelProps<"Silver">
+  | TierTwoLevelProps<"Bronze">
```

Comment:
These sets of changes were the brain child of @joshuayoes & I in a pairing session. 🤩